designdocs/capacity_staking_rewards_implementation.md

```diff
@@ -0,0 +1,324 @@
+# Capacity Staking Rewards Implementation
+The proposed feature is a design for staking FRQCY token in exchange for Capacity and/or FRQCY.
+It is specific to the Frequency Substrate parachain.
+It consists of enhancements to the capacity pallet, needed traits and their implementations, and needed runtime configuration.
+
+This does _not_ outline the economic model for Staking Rewards (also known as "Provider Boosting"); it describes the economic model as a black box, i.e. an interface.
+
+## Context and Scope:
+The Frequency Transaction Payment system uses Capacity to pay for certain transactions on chain.  Accounts that wish to pay with Capacity must:
+
+1. Have an [MSA](https://github.com/LibertyDSNP/frequency/blob/main/designdocs/accounts.md)
+2. Be a [Provider](https://github.com/LibertyDSNP/frequency/blob/main/designdocs/provider_registration.md) (see also [Provider Permissions and Grants](https://github.com/LibertyDSNP/frequency/blob/main/designdocs/provider_permissions.md))
+3. Stake a minimum amount of FRQCY (on mainnet, UNIT on Rococo testnet) token to receive [Capacity](https://github.com/LibertyDSNP/frequency/blob/main/designdocs/capacity.md).
+
+# Problem Statement
+This document outlines how to implement the Staking for Rewards feature described in [Capacity Staking Rewards Economic Model (TBD)](TBD), without, at this time, regard to what the economic model actually is.
+
+## Glossary
+1. **FRQCY**: the native token of the Frequency blockchain
+1. **Capacity**: the non-transferrable utility token which can be used only to pay for certain Frequency transactions.
+1. **Account**: a Frequency System Account controlled by a private key and addressed by a public key, having at least a minimum balance (currently 0.01 FRQCY).
+1. **Stake** (verb): to lock some amount of a token against transfer for a period of time in exchange for some reward.
+1. **RewardEra**: the time period (TBD in blocks or Capacity Epochs) that Staking Rewards are based upon. RewardEra is to distinguish it easily from Substrate's staking pallet Era, or the index of said time period.
+1. **Staking Reward**: a per-RewardEra share of a staking reward pool of FRQCY tokens for a given staking account.
+1. **Reward Pool**:  a fixed amount of FRQCY that can be minted for rewards each RewardEra and distributed to stakers.
+1. **StakingRewardsProvider**: a trait that encapsulates the economic model for staking rewards, providing functionality for calculating the reward pool and staking rewards.
+
+## Staking Token Rewards
+
+### StakingAccountDetails updates
+New fields are added. The field `last_rewarded_at` is to keep track of the last time rewards were claimed for this Staking Account.
+MaximumCapacity staking accounts MUST always have the value `None` for `last_rewarded_at`.  This should be the default value also.
+`MaximumCapacity` is also the default value for `staking_type` and should map to 0.
```

Comment:
I think it might be easier to follow if you introduce the different `StakingType` or any other types like that first.